decode.go

```diff
@@ -29,6 +29,22 @@ func NewDecoder(r io.Reader) *Decoder {
 	return dec
 }
 
+// NewDecoderSize returns a new decoder that reads from r.
+//
+// The decoder introduces its own buffering and may read data from r beyond
+// the logfmt records requested.
+// The size argument specifies the size of the initial buffer that the
+// Decoder will use to read records from r.
```

Comment:
Thanks for including a good doc comment. I think it is important to also make it clear that ScanRecord will fail for any line longer than size.